blogs/2022/09/15/dev-container-features.md

```diff
@@ -0,0 +1,139 @@
+---
+Order: 76
+TOCTitle: Dev Container Features
+PageTitle: Dev Container Features
+MetaDescription: What are dev container Features and what's new
+Date: 2022-09-15
+Author: Brigit Murtaugh
+---
+
+# Dev Container Features
+
+September 15, 2022 by Brigit Murtaugh, [@BrigitMurtaugh](https://twitter.com/BrigitMurtaugh)
+
+We’ve all had that moment when setting up our development environment – "Oh, I just need one more thing!" – that "thing" being one more language or toolset (or maybe a few more 😊) to work on your project.
+
+Development containers are a great way to simplify environment set up - they provide a full-featured coding environment with the tools your project needs. They’re configured using an image, Dockerfile, or Docker Compose file and `devcontainer.json`, which is a metadata format used to enrich containers with development specific content and settings.
+
+When creating a dev container, you may encounter the same "I just need one more thing!" – maybe you’re using a Node.js image in your Dockerfile and just need to add Git. Or maybe you need to add something more complex, like working with Docker or Kubernetes from within your dev container. Dev containers are great since anyone accessing your code will have the same, consistent experience with all those tools you added – but what’s the best way to add them?
+
+What if there was an easy way to install that extra tool in your dev container, simply by mentioning the tool’s name and version? Or what if as a tool user or author, you could create an easy way for others to install it? Sharing manual scripts can help with reuse, but when referencing one, you may forget to reference container or tooling settings, such as enabling ptrace support for Go, Rust, or C++ debugging, adding a specific entry point to fire on container start, or ensuring the right VS Code extensions are included.
+
+We’re happy to share that dev container feature packs help you smoothly get the tools you need in your dev container!
+
+Feature packs are self-contained units of installation code, designed to install new features in your dev container atop a wide-range of base container images. As part of our work on the [open dev container specification](https://containers.dev/), we’ve made some improvements to where you can grab pre-created feature packs and how you can author and distribute your own.
+
+Let’s see what’s new and how you can get started with features from any dev container supporting tool or service (like the [VS Code Remote - Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) or [GitHub Codespaces](https://github.com/features/codespaces))!
+
+## Using Features
```

Comment:
Perhaps "Using Feature Packs?"  Or "Installing Features?"

Reply:
I thought the differentiation would be: 

- You install a feature pack
- You use a feature(s) from the pack

Sounds like you're thinking the opposite?

Reply:
Yeah, that works too - but isn't this section focusing on "Installing feature packs" with that terminology? (Verses using them once installed?) Maybe that's the right heading then.

Reply:
I like `Adding features to your dev container` - updating now.